### PR TITLE
MongoDB Criteria API

### PR DESCRIPTION
**Pull request by amoscatelli**
This is a partial implementation of the Criteria API for MongoDB
I made an early pull request as a proof of concept of the Criteria API

@otaviojava 

as you can see, the API is not specific for MongoDB
I added a method in the DocumentCollectionManager

Do you see now ?
If you still want me to convert CriteriaQuery into a DocumentQuery please tell me more on how you imagine it

### CODE REVIEW

**Comment by genie-jnosql** [comment]
Can one of the admins verify this patch?

**Comment by otaviojava** [comment]
@amoscatelli I saw it now thank you.
It is a great future for the community.

File: mongodb-driver/src/test/java/org/eclipse/jnosql/communication/mongodb/document/MongoDBDocumentCollectionManagerTest.java [hunk 1]
```diff
@@ -513,6 +520,390 @@ public void shouldSaveMap() {
...
+        entityManager.insert(getEntity());
+        CriteriaQuery<Person> criteriaQuery = new DefaultCriteriaQuery(Person.class);
+        SelectQuery<Person> select = criteriaQuery.select();
+        assertTrue(entityManager.executeQuery(select).toArray().length == 1);
+    }
+
+    @Test
+    public void shouldSelectByAttribute() {
+        entityManager.insert(getEntity());
+        CriteriaQuery<Person> criteriaQuery = new DefaultCriteriaQuery(Person.class);
```
Comment (otaviojava) [inline mongodb-driver/src/test/java/org/eclipse/jnosql/communication/mongodb/document/MongoDBDocumentCollectionManagerTest.java:534]: the code, in general, looks good, but, the communication layer should not have access to an entity class such as ```Person```

Comment (otaviojava) [reply mongodb-driver/src/test/java/org/eclipse/jnosql/communication/mongodb/document/MongoDBDocumentCollectionManagerTest.java:534]: I believe if we only move it as either a mapping class or include it as an extension:

https://github.com/eclipse/jnosql-mapping-extension/tree/master/mongodb-extension

**Comment by amoscatelli** [comment]
I am going to close this since we agreed to convert Criteria API into DocumentQuery to keep it for mapping layer only

**Pull request closed without merging**